### PR TITLE
add shellcheck option to the doc and sort options by theme

### PR DESCRIPTION
**Pull request by Geam**
I've sort the options cause I find it quite messy to have option for same language not one after one other. If you don't want it, I can reset to commit where I add the shellcheck option.

Sorry I was not available as much as I wanted this week-end for PR #66.

### CODE REVIEW

File: doc/ale.txt [hunk 1]
```diff
@@ -393,41 +407,40 @@ g:ale_fortran_gcc_options                           *g:ale_fortran_gcc_options*
...
+4.9. shellchek                                  *ale-linter-options-shellcheck*
 
-g:ale_javascript_jshint_executable         *g:ale_javascript_jshint_executable*
+g:ale_linters_sh_shellckeck_exclusions *g:ale_linters_sh_shellckeck_exclusions*
 
   Type: |String|
-  Default: `'jshint'`
+  Default: `''`
+
+  Set this variable to exclude test for shellcheck (-e option);
```
Comment (w0rp) [inline doc/ale.txt]: Should that read "exclude tests?" It looks like that's intended to be plural.

**Comment by w0rp** [comment]
I saw this kind of sorting coming. The headings might get sorted again in future, possibly some different way. I'm pretty happy with what you've done here.

File: doc/ale.txt (hunk 1 above)
Comment (w0rp) [inline doc/ale.txt]: Also, it looks like there's a random semicolon on the end of the line.

Comment (Geam) [inline doc/ale.txt]: Woopsie the semi-column, too much c/cpp latetly.
For the -e as I'm not a shellcheck user I've report to the doc and I don't think it's suppose to be plural : https://github.com/koalaman/shellcheck/wiki/Ignore
I'll dig through that a bit more.

Comment (Geam) [inline doc/ale.txt]: Ok it's suppose to be plural : https://github.com/koalaman/shellcheck/issues/14

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**